File: admin-api/models/audit.py
from typing import Literal

from pydantic import BaseModel


class AuditLogEntry(BaseModel):
    """Matches frontend AuditLogEntry exactly."""

    id: int
    timestamp: str
    key: str
    old_value: str
    new_value: str
    source: Literal["user", "system"]
    user: str

File: admin-panel/types/audit.ts
export interface AuditLogEntry {
  id: number;
  timestamp: string;
  key: string;
  old_value: string;
  new_value: string;
  source: "user" | "system";
  user: string;
}

export interface AuditFilters {
  category: string;
  source: string;
  dateFrom?: string;
  dateTo?: string;
}
